From 09598895c607f3e41f0249ade217ace0a4da6437 Mon Sep 17 00:00:00 2001
From: Phil Thompson <phil@riverbankcomputing.com>
Date: Sun, 23 Aug 2026 12:18:00 +0100
Subject: [PATCH] Bug fixes

- Fixed a regression in SIP v6.16.1 that meant that bindings using legacy
ABIs would not build if they didn't specify `%MinimumABIVersion`.
Resolves #114
---
 sipbuild/generator/parser/parser_manager.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sipbuild/generator/parser/parser_manager.py b/sipbuild/generator/parser/parser_manager.py
index 8f2501ff..83c63310 100644
--- a/sipbuild/generator/parser/parser_manager.py
+++ b/sipbuild/generator/parser/parser_manager.py
@@ -1938,6 +1938,10 @@ def _finalise_abi_version(self):
                 # We have no minimum version information from which to derive a
                 # major version so use the default.
                 major = DEFAULT_ABI_MAJOR
+        elif major <= 13 and not minimums:
+            # Allow older ABIs that don't specify %MinimumABIVersion.  This
+            # case can be removed in SIP v7.
+            pass
         else:
             # Check that this major version is supported and that any minor
             # version meets the minimum requirements.
